src/build/env.py

```diff
@@ -177,35 +266,56 @@ def install(self, requirements: Collection[str]) -> None:
                 os.path.abspath(req_file.name),
             ]
             run_subprocess(cmd)
+
         finally:
             os.unlink(req_file.name)
 
+    @property
+    def name(self) -> _EnvImplInclDefault:
+        return 'virtualenv+pip' if self._create_with_virtualenv else 'venv+pip'
 
-def _create_isolated_env_virtualenv(path: str) -> tuple[str, str]:
-    """
-    We optionally can use the virtualenv package to provision a virtual environment.
 
-    :param path: The path where to create the isolated build environment
-    :return: The Python executable and script folder
-    """
-    import virtualenv
+class _UvImplBackend(_EnvImplBackend):
+    def create(self, path: str) -> None:
+        import venv
 
-    if _valid_global_pip():
-        cmd = [path, '--no-seed', '--activators', '']
-    else:
-        cmd = [path, '--no-setuptools', '--no-wheel', '--activators', '']
+        self._env_path = path
+
+        # ``uv.find_uv_bin`` will look for uv in the user prefix if it can't
+        # find it under ``sys.prefix``, essentially potentially rearranging
+        # the user's $PATH.  We'll only look for uv under the prefix of
+        # the running interpreter for unactivated venvs then defer to $PATH.
+        uv_bin = shutil.which('uv', path=sysconfig.get_path('scripts'))
```

Comment:
uv comes with a find_uv_bin method we should use that.

Reply:
The comment explains why it's not being used.

Reply:
Guess I agree here with uv's choice, I find it more expected, and disagree with the comment.

Reply:
How is it expected that a uv you've installed in a venv might load a uv from an outer env?

```
$ python -m venv --without-pip foo
$ VIRTUAL_ENV=foo uv pip install uv
Resolved 1 package in 3ms
Installed 1 package in 5ms
 + uv==0.1.16
$ foo/bin/python -c 'import uv; print(uv.find_uv_bin())'
/Users/dimitris/code/python-packaging/build/foo/bin/uv
$ rm foo/bin/uv
$ foo/bin/python -c 'import uv; print(uv.find_uv_bin())'
/Users/dimitris/.local/bin/uv
```

If uv should function as a Python package, it has no business looking outside its prefix.

I'd personally prefer to take a standards-based approach than rely on `uv`'s custom script finder.  Is this not the correct way to locate a script under the running interpreter's prefix?

Reply:
Open an issue in UV and we should fix it there then.

Reply:
Chaned to only use `uv.find_uv_bin`.